pkg/util/aadapp/aadapp.go

```diff
@@ -25,7 +23,27 @@ func UpdateAADAppSecret(ctx context.Context, appClient azureclient.RBACApplicati
 	if len(apps) != 1 {
 		return "", fmt.Errorf("found %d applications, should be 1", len(apps))
 	}
-	app := apps[0]
+	return *apps[0].ObjectID, nil
+}
+
+// GetObjectIDUsingSPClient find the ObjectID for the application
+func GetObjectIDUsingSPClient(ctx context.Context, spc azureclient.ServicePrincipalsClient, appID string) (string, error) {
```

Comment:
@jim-minter I tried using either of these two functions for both use cases and kept getting 403's. It was driving me crazy maybe someone can figure it out? at least they are together so we know about them...

Reply:
2 different objects - AAD applications (e.g. `az ad app list`) and service principals (e.g. `az ad sp list`).  Each have an object ID; each also have an appID field.

Reply:
So my advice that the same function could be used for both was wrong - sorry.